llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -479,27 +480,78 @@ SUnit *AIEPostRASchedStrategy::pickNodeAndCycle(
 }
 
 int AIEPostRASchedStrategy::getMaxDeltaCycles(const SchedBoundary &Zone) const {
-  assert(!Zone.isTop());
-  if (Zone.getCurrCycle() >= RegionBottomUpCycles - 1)
+  // Top-down scheduling does not support DeltaCycles
+  if (Zone.isTop() || Zone.getCurrCycle() >= RegionBottomUpCycles - 1)
     return 0;
   return std::min({int(RegionBottomUpCycles - 1 - Zone.getCurrCycle()),
                    int(getAIEHazardRecognizer(Zone)->getMaxLookAhead()),
                    BottomUpDelta.getValue()});
 }
 
+unsigned getNumEmittedInstrs(ScheduleDAGMI *DAG, bool IsTop) {
```

Comment:
I could do with some description. I also hope that std::distance is cheap here.

Reply:
`/// Returns the number of emitted instructions in the Top or Bot zone`?